BuildingBlocks/src/BuildingBlocks.Infrastructure/Persistence/BlobStorage/BlobStorageServiceCollectionExtensions.cs

```diff
@@ -24,13 +26,18 @@ public static void AddBlobStorage(this IServiceCollection services, BlobStorageO
         {
             case AZURE_CLOUD_PROVIDER:
                 services.AddAzureStorageAccount(azureStorageAccountOptions => { azureStorageAccountOptions.ConnectionString = options.ConnectionInfo!; });
+                services.AddHealthChecks().AddAzureBlobStorage(
+                    clientFactory: _ => new BlobServiceClient(options.ConnectionInfo),
+                    optionsFactory: _ => new AzureBlobStorageHealthCheckOptions { ContainerName = options.Container }
+                );
```

Comment:
Can you please check whether the result of the clientFactory is cached somehow or whether a new BlobServiceClient is created on every health check?
Because if it's the latter, we need a different solution.

Reply:
The client factory is created when a `AzureBlobStorageHealthCheck` instance is created. It is created with a factory during the creation of a `HealthCheckRegistration`. I don't think that the registration is recreated at every health check, but I'm not sure if the Health check class is created multiple times.

Reply:
Testing this should be as simple as setting a breakpoint into the client factory and running the health check a bunch of times.

Reply:
The client factory gets created every time, but it can be configured to use the service provider, so I added the `BlobServiceClient` as a Singleton.